core/domain/email_manager_test.py

```diff
@@ -2051,7 +2055,2981 @@ def test_that_correct_completion_email_is_sent(self):
+        }
+
+    def test_email_not_sent_if_can_send_emails_is_false(self):
+        config_services.set_property(
+            'committer_id', 'notify_admins_reviewers_needed_is_enabled', True)
+
+        with self.cannot_send_emails_ctx, self.log_new_error_ctx:
+            email_manager.send_mail_to_notify_admins_reviewers_needed(
+                [self.admin_1_id], self.suggestion_types_need_reviewers)
+
+        messages = self._get_all_sent_email_messages()
+        self.assertEqual(len(messages), 0)
+        self.assertEqual(self.log_new_error_counter.times_called, 1)
+        self.assertEqual(
+            self.logged_errors[0], 'This app cannot send emails to users.')
+
+    def test_email_not_sent_if_notifying_admins_reviewers_needed_is_disabled(
+            self):
+        config_services.set_property(
+            'committer_id', 'notify_admins_reviewers_needed_is_enabled', False)
+
+        with self.can_send_emails_ctx, self.log_new_error_ctx:
+            email_manager.send_mail_to_notify_admins_reviewers_needed(
+                [self.admin_1_id], self.suggestion_types_need_reviewers)
+
+        messages = self._get_all_sent_email_messages()
+        self.assertEqual(len(messages), 0)
+        self.assertEqual(self.log_new_error_counter.times_called, 1)
+        self.assertEqual(
+            self.logged_errors[0],
+            'The "notify_admins_reviewers_needed_is_enabled" property '
+            'must be enabled on the admin config page in order to send '
+            'admins the emails.')
+
+    def test_email_not_sent_if_no_admins_to_notify(self):
+        config_services.set_property(
+            'committer_id', 'notify_admins_reviewers_needed_is_enabled', True)
+
+        with self.can_send_emails_ctx, self.log_new_error_ctx:
+            email_manager.send_mail_to_notify_admins_reviewers_needed(
+                [], self.suggestion_types_need_reviewers)
+
+        messages = self._get_all_sent_email_messages()
+        self.assertEqual(len(messages), 0)
+        self.assertEqual(self.log_new_error_counter.times_called, 1)
+        self.assertEqual(
+            self.logged_errors[0],
+            'No admins to notify that Contributor Dashboard reviewers are '
+            'needed.')
+
+    def test_email_not_sent_if_no_suggestion_types_that_need_reviewers(
+            self):
+        config_services.set_property(
+            'committer_id', 'notify_admins_reviewers_needed_is_enabled', True)
+
+        with self.can_send_emails_ctx, self.log_new_error_ctx:
+            email_manager.send_mail_to_notify_admins_reviewers_needed(
+                [self.admin_1_id], {})
+
+        messages = self._get_all_sent_email_messages()
+        self.assertEqual(len(messages), 0)
+        self.assertEqual(self.log_new_error_counter.times_called, 1)
+        self.assertEqual(
+            self.logged_errors[0],
+            'There were no suggestion types that needed more reviewers on the '
```

Comment:
Should this really be an error case? I sort of expect the cron job just to maybe log that there are no types to check, but it's not really a failure case, is it? It's actually the ideal state.

Reply:
I changed the errors to infos